### ISSUE TITLE
Arrays of length 1 are compacted - IMO they shouldn't be

### ISSUE THREAD

**Issue by davidkalosi** [seznam/halson#6]
this test case is IMO not correct since it should retain the array type if the embed is inserted as arrays. I am running into this issue right now - If there is only one item in the array it is turned into an object which is not correct when the input is an array. 

```
it('add first embed array', function() {
            var res = halson();
            var embed = [{title: "Untitled"}];
            var expected = {
                _embedded: {
                    item: {
                        title: "Untitled"
                    }
                }
            };

            res.addEmbed('item', embed);
            assert.deepEqual(res, expected);
        });
```

the embed should look like 
```
var expected = {
                _embedded: {
                    item: [{
                        title: "Untitled"
                    }]
                }
            };
```

**Comment by davidkalosi** [comment]
a PR with the fix is ready. all tests are passing.

**Comment by hajovsky** [comment]
Why it can't be automatically compacted? The result is still a valid hal+json format. I think you should be able to work with both formats. 

I don't see a reason to suspend auto-compact feature in this particular case.

**Comment by davidkalosi** [comment]
For example when using the square retrofit rest client - you have 

public class Cart {
  private List<CartItem> items;
}

Now this will fail if you have a cart with one item because it won't be an array anymore. It would have the following java representation

public class Cart {
  private CartItem items;
}

and thats not good because the relationship should be 1..n even if there is a single record.

**Comment by chosecz** [comment]
I agree with David. It's good to keep array even with single record.

2015-01-13 15:22 GMT+01:00 David Kalosi <notifications@github.com>:

**Comment by davidkalosi** [comment]
@hajovsky have you made a decision on this one ? 
thanks david

**Comment by hajovsky** [comment]
@davidkalosi I'll fix this issue later today.

**Comment by davidkalosi** [comment]
supercool ;) the PR is ready you can review it but it's just a small fix. all the tests are passing as well so you can just merge it. 

thanks
david

**Comment by hajovsky** [comment]
Fixed in #7
